camunda-engine-rest-client-openapi-springboot/src/main/java/org/camunda/community/rest/client/springboot/CamundaProcessAutodeployment.java

```diff
@@ -72,7 +73,8 @@ private void deployResources(List<Resource> resourcesToDeploy, String type) thro
             // but the OpenAPI will need a File.
             // We still have to set the file ending correct in the temp file
             // (because otherwise the deployer will not pick it up as e.g. BPMN file)
-            final File tempFile = File.createTempFile(UUID.randomUUID().toString(),"." + type);
+            String tempDirectoryName = FileUtils.getTempDirectory().getAbsolutePath();
+            final File tempFile = new File(tempDirectoryName + File.separator + camundaResource.getFilename());
```

Comment:
The problem here is, that "camundaResource.getFilename" can be null if some contexts - that's why I introduced the temp file name in the first place (https://github.com/camunda-community-hub/camunda-platform-7-rest-client-java/commit/d3ddc3d30eb2801dcdcdf251f9fd85f1aa27a3e5).

Reply:
well, if camundaResource.getFilename() == "" it means that it is not a FileSystemResource,
but all camunda resources are files in your projects, they are FileSystemResource instances.

for example I've  io.github.openfeign.form dependency, and got " io.github.openfeign.form/"  resource.
(there are  UrlResource instances)

So we should  check that resources are instances of FileSystemResource